DMTN-306.tex

```diff
@@ -1,79 +1,135 @@
+
+Each processing facility exposes at least two RSEs: one for storing input data required for processing (e.g., raw images, calibration data, reference catalogs) and another endpoint for storing the products generated by the image processing pipelines. All RSEs are configured to use the identity logical-to-physical filename mapping. This configuration ensures that the file pathnames are preserved relative to the Butler repository's datastore location, which is critical for proper file replication to the destination where the Butler expects to find them.
+
+\subsection{Registration of files to replicate}
+\label{registration}
+
+To perform replication, we create Rucio Datasets, each composed of a set of files that are already in their appropriate locations on the source RSE. Upon registration, pre-configured Rucio subscriptions trigger the actual file movement to the destination facility, in accordance with the defined replication rules. Rucio delegates the data transfer to FTS, which then instructs the storage endpoints at the facilities to move the data, typically by requesting the destination facility to pull the data from the source facility.
+
+Rubin has developed the tool \texttt{rucio\_register}\footnote{https://github.com/lsst/rucio\_register}, which allows for the selection of existing files from a Butler repository based on specified criteria. The tool attaches Rubin-specific metadata to these files and registers them into one or more Rucio Datasets. The metadata, encoded as a JSON record, contains a minimal set of information extracted from the origin Butler repository. This ensures that replicated files can be ingested properly into the local Butler repository at the destination facility.
+
+Only files that require replication to another facility are registered with Rucio. As a result, Rubin’s instance of Rucio is aware only of files replicated across processing facilities. Files that are local to each facility and not subject to replication remain known only to that facility's Butler and are not registered in Rucio.
+
+\subsection{Ingestion at reception}
+\label{ingestion}
+
+FTS notifies Rucio about the completion of individual file transfers. Rubin's \texttt{ctrl\_rucio\_ingest}\footnote{https://github.com/lsst-dm/ctrl\_rucio\_ingest}, an extension to Rucio's Hermes daemon, uses Kafka as a mechanism to signal the destination Rubin facility that a new file was replicated and to take appropriate actions.
+
+Messages distributed through Rubin's Kafka control-plane include Rubin-specific metadata. Those messages are received by Rubin's \texttt{ctrl\_ingestd}\footnote{https://github.com/lsst-dm/ctrl\_ingestd}, a daemon running at each destination facility responsible for ingesting newly replicated files into the local Butler repository.
```

Comment:
Messages distributed through Rubin's Kafka control-plane include Rubin-specific metadata. Those messages are received by Rubin's \texttt{ingestd}\footnote{https://github.com/lsst-dm/ctrl\_ingestd}, a daemon running at each destination facility responsible for ingesting newly replicated files into the local Butler repository.

Reply:
Implemented